Dockerfile

```
@@ -1,8 +1,9 @@
 # syntax=docker/dockerfile:1
 FROM golang:1.18 as builder
+ARG VERSION
 WORKDIR /build
 ADD . /build/
-RUN --mount=type=cache,target=/root/.cache/go-build make build-for-docker
+RUN --mount=type=cache,target=/root/.cache/go-build CGO_ENABLED=0 GOOS=linux go build -trimpath -ldflags "-s -X main.version=${VERSION}" -v -o your-project main.go
 
 FROM scratch
```

lets use alpine bcz most of our project requires secret which needs shell access.
```suggestion
FROM alpine:latest
```